examples/src/main/scala/ContextStorageExample.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import cats.effect.IO
+import cats.effect.IOApp
+import cats.effect.Resource
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ContextKey
+import io.opentelemetry.context.ContextStorage
+import org.typelevel.otel4s.java.IOLocalContextStorageProvider
+
+import java.util.logging._
+
+object ContextStorageExample extends IOApp.Simple {
+
+  val key = ContextKey.named[String]("test")
+
+  val printKey =
+    IO(Option(Context.current().get(key))).flatMap(v => IO.println(v))
+
+  def run =
+    for {
+      _ <- IO {
+        val rootLog = Logger.getLogger("")
+        rootLog.setLevel(Level.FINE)
+        rootLog.getHandlers().head.setLevel(Level.FINE)
+      }
+      _ <- IOLocalContextStorageProvider.localContext.set(Context.root())
```

Comment:
This line is tasteless, but without it:

```
[error] java.util.NoSuchElementException: key not found: cats.effect.IOLocal$IOLocalImpl@63913f7c
[error] 	at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:223)
[error] 	at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:219)
[error] 	at apply @ ContextStorageExample$.$anonfun$run$3(ContextStorageExample.scala:44)
[error] 	at main$ @ ContextStorageExample$.main(ContextStorageExample.scala:27)
[error] 	at main$ @ ContextStorageExample$.main(ContextStorageExample.scala:27)
[error] 	at main$ @ ContextStorageExample$.main(ContextStorageExample.scala:27)
[error] 	at flatMap @ ContextStorageExample$.$anonfun$run$3(ContextStorageExample.scala:43)
[error] 	at flatMap @ ContextStorageExample$.$anonfun$run$2(ContextStorageExample.scala:42)
[error] 	at apply @ ContextStorageExample$.run(ContextStorageExample.scala:36)
[error] 	at flatMap @ ContextStorageExample$.run(ContextStorageExample.scala:36)
[error] 	at run$ @ ContextStorageExample$.run(ContextStorageExample.scala:27)
```

Reply:
Sorry, my fault. I haven't done the work to support default values yet, that will fix this.

Reply:
Just pushed some improvements to my CE PR, that should fix this, as of https://github.com/typelevel/cats-effect/commit/d55489d8e442fe9cdac4178db39792bb3cdb4f43.